src/Shared.Core/Utilities/ArrayUtils.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace KKAPI.Utilities
+{
+    /// <summary>
+    /// Utilities for arrays
+    /// </summary>
+    /// <inheritdoc />
+    public class ArrayUtils
+    {
+        /// <summary>
+        /// Compares two byte arrays for equality in a high-performance manner using unsafe code.
+        /// </summary>
+        /// <param name="a">The first byte array to compare.</param>
+        /// <param name="b">The second byte array to compare.</param>
+        /// <returns>True if the byte arrays are equal, false otherwise.</returns>
+        static public bool SequenceEqualFast(byte[] a, byte[] b)

```

Comment:
I suggest changing this into an extension method and moving it to Utilities/Extensions.cs
It can also be made public to allow other plugins to use it.

Reply:
Fixed! Please review